src/Hosting/Hosting/src/HostBuilder.cs

```diff
@@ -187,6 +187,9 @@ private void BuildAppConfiguration()
             }
             _appConfiguration = configBuilder.Build();
             _hostBuilderContext.Configuration = _appConfiguration;
+
+            // dispose host configuration since it is no longer accessible
```

Comment:
Too soon, the host config is chained into the app config. You have to wait until host disposal.

Reply:
Oh. I was under the impression that `configurationBuilder.AddConfiguration(IConfiguration)` would actually _copy_ over the values from the source configuration. But it does actually link them. TIL!

Would it work for you to make the `ChainedConfigurationProvider` disposable and dispose of the chained configuration then? That would actually make this much nicer too.

Reply:
I went with that idea now and made it configurable to avoid breaking any existing code that would use the same configuration multiple times.

In the framework, the only places `AddConfiguration` is actually being used outside of the host builders are adding configuration sections which aren’t disposable and won’t trigger this. So maybe making the dispose configurable isn’t actually required.